src/components/Team/index.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { useState } from 'react';
+import Select from 'react-select';
+
+import TeamCard from './TeamCard';
+
+interface Person {
+  name: string;
+  class: number;
+  roles: string[];
+  image?: string;
+}
+
+interface TeamProps {
+  data: Person[];
+}
+
+const roles = ['All Roles', 'Designer', 'Developer', 'Director', 'External', 'Projects'];
+const roleOptions = roles.map(item => ({ value: item, label: item }));
+
+const currYear = new Date().getFullYear();
+const yearOptions = ['All Years', currYear + 1, currYear + 2, currYear + 3, currYear + 4, 'Alumni'].map(item => ({ value: item, label: item }));
+
+const selectStyles = {
+  control: (provided, _) => ({
+    ...provided,
+    borderRadius: '0',
+    padding: '8px 4px 8px 8px',
+  }),
+  menu: (provided, _) => ({
+    ...provided,
+    borderRadius: '0',
+  }),
+  dropdownIndicator: (provided, _) => ({
+    ...provided,
+    color: 'black',
+  }),
+  indicatorSeparator: (provided, _) => ({
+    ...provided,
+    display: 'none',
+  }),
+};
+
+function Team(props: TeamProps): JSX.Element {
+  const [year, setYear] = useState({ value: 'All Years', label: 'All Years' });
+  const [role, setRole] = useState({ value: 'All Roles', label: 'All Roles' });
+
+  // if there's no image, use 'winter.svg'
+  props.data.forEach((data) => (data.image && data.image.indexOf('/assets/') == -1 ? (data.image = `/assets/${data.image}`) : data.image) || (data.image = '/assets/winter.svg'));
+
+  let filteredData = props.data;
+
+  if (year.value !== 'All Years') {
+    if (year.value === 'Alumni') {
+      filteredData = filteredData.filter((person) => person.class <= currYear);
+    }
+    else {
+      filteredData = filteredData.filter((person) => person.class === year.value);
+    }
+  }
+
+  if (role.value !== 'All Roles') {
+    filteredData = filteredData.filter((person) => person.roles.includes(role.value));
+  }
```

Comment:
I think it would be better if we turned `filteredData` into another state variable and then used `useEffect` hooks to filter it based on the value of `year` and `role`. It isn't a huge deal since `props.data` is always gonna be pretty small, but this way it would avoid recalculating the filtered data on every rerender and is just good practice.

(This suggestion might not work completely but thought I'd save you a little bit of trouble:) 
 ```suggestion
   props.data.forEach((data) => (data.image && data.image.indexOf('/assets/') == -1 ? (data.image = `/assets/${data.image}`) : data.image) || (data.image = '/assets/winter.svg'));

   const [filteredData, setFilteredData] = useState<Person[]>(props.data);

   useEffect(() => {
     if (year.value === 'Alumni') {
       setFilteredData(oldFilteredData => oldFilteredData.filter((person) => person.class <= currYear));
     } else if (year.value !== 'All years') {
       setFilteredData(oldFilteredData => oldFilteredData.filter((person) => person.class === year.value));
     }
   }, [ year ]);
  
   useEffect(() => {
     if (role.value !== 'All Roles') {
       setFilteredData(oldFilteredData => oldFilteredData.filter((person) => person.roles.includes(role.value)));
     }
   }, [ role ]);
```